frontend/javascripts/oxalis/model/sagas/mapping_saga.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import _ from "lodash";
+import {
+  type Saga,
+  _takeLatest,
+  call,
+  select,
+  take,
+  put,
+  _actionChannel,
+} from "oxalis/model/sagas/effect-generators";
+import { message } from "antd";
+
+import {
+  setMappingAction,
+  setMappingEnabledAction,
+  type SetMappingAction,
+} from "oxalis/model/actions/settings_actions";
+import { fetchMapping } from "admin/admin_rest_api";
+import type { APIMapping } from "types/api_flow_types";
+import { getLayerByName } from "oxalis/model/accessors/dataset_accessor";
+import { type Mapping } from "oxalis/store";
+import ErrorHandling from "libs/error_handling";
+import { MAPPING_MESSAGE_KEY } from "oxalis/model/bucket_data_handling/mappings";
+
+type APIMappings = { [string]: APIMapping };
+
+export default function* watchActivatedMappings(): Saga<void> {
+  // Buffer actions since they might be dispatched before WK_READY
+  const actionChannel = yield _actionChannel("SET_MAPPING");
+  yield* take("WK_READY");
+  yield _takeLatest(actionChannel, maybeFetchMapping);
+}
+
+function* maybeFetchMapping(action: SetMappingAction): Saga<void> {
+  const { layerName, mappingName, mappingType, mapping: existingMapping } = action;
+
+  if (mappingName == null || existingMapping != null) {
+    return;
+  }
+
+  if (mappingType !== "JSON") {
+    // JSON mappings will be activated once they have been fetched
```

Comment:
the comment reads a bit weird since the if-branch refers to the non-json case. can you tune the comment a bit?